lib/src/react_client/synthetic_data_transfer.dart

```diff
@@ -104,8 +104,8 @@ SyntheticDataTransfer syntheticDataTransferFactory(Object dt) {
   }
 
   // Copy these lists and ensure they're typed properly.
-  final files = <File>[...?rawFiles];
-  final types = <String>[...?rawTypes];
+  final files = List<File>.from(rawFiles ?? []);
+  final types = List<String>.from(rawTypes ?? []);
```

Comment:
Shouldn't this be

```suggestion
  final files = List.from(rawFiles ?? []).cast<File>();
  final types = List.from(rawTypes ?? []).cast<String>();
```

Reply:
I think in this case, either would be acceptable. 

`List<File>.from(rawFiles ?? [])` creates a new list and casts objects to the target type as they're copied.

`List.from(rawFiles ?? []).cast<File>()` creates a new `List<dynamic>` and then returns a view on that list that casts items as they're accessed.

Casting them on copy feels better to me in this case, though, so that any bad typing is revealed always, when the event is dispatched, as opposed to only when these properties are accessed.